node/src/reactor/joiner.rs

```diff
@@ -518,8 +518,8 @@ impl reactor::Reactor for Reactor {
                                 block_header_with_old_version,
                             }) => {
                                 let old_version = block_header_with_old_version.protocol_version();
-                                info!(%current_version, %old_version, "restarting for downgrade");
-                                Some(JoinerEvent::Shutdown(ExitCode::DowngradeVersion))
+                                error!(%current_version, %old_version, "downgrade is not allowed, exiting");
+                                panic!("downgrade is not allowed");
```

Comment:
We should use the `fatal!` macro/effect instead of panicking.

Reply:
Fixed here: https://github.com/casper-network/casper-node/pull/2434/commits/634c0c8f110c69fae8cb782356874ae7ce4fd5c8